### ISSUE TITLE
[firebase_messaging]: v15.1.4 APNS token has not been set yet. Please ensure the APNS token is available by calling `getAPNSToken()`

### ISSUE THREAD

**Issue by ziqq** [firebase/flutterfire#15384]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Which plugins are affected?

Messaging

### Which platforms are affected?

iOS

### Description

Not working after upgrade. I make clean and remove Podfile.lock.

1. [Request](https://github.com/firebase/flutterfire/pull/13570#issuecomment-2467338592)
2. [Issue](https://github.com/firebase/flutterfire/issues/13558)

### Firebase Core version

3.7.0

### Flutter Version

3.24.4

### Relevant Log Output

```shell
APNS token has not been set yet. Please ensure the APNS token is available by calling `getAPNSToken()`
```

### Flutter dependencies

<details>
<summary>Expand <code>Flutter dependencies</code> snippet</summary>
<br>

```yaml
Dart SDK 3.5.4
Flutter SDK 3.24.4

dependencies:
- appmetrica_plugin 3.1.0 [flutter stack_trace logging decimal]
- async 2.11.0 [collection meta]
- auto_route 9.2.2 [flutter path web collection meta]
- bb_core 2.4.0 [flutter flutter_localizations intl http flutter_secure_storage shared_preferences async path meta collection path_provider url_launcher]
- bb_http_client 3.9.0 [flutter flutter_localizations intl collection convert async meta path dio dio_smart_retry retry sentry_flutter l stack_trace bb_core]
- bb_ui_kit 3.1.0 [flutter flutter_localizations intl path meta async convert equatable collection platform_info url_launcher permission_handler mask_text_input_formatter flutter_svg flutter_slidable flutter_rating_bar flutter_inset_box_shadow cupertino_icons another_flushbar image_picker extended_image carousel_slider sticky_headers pull_down_button cached_network_image custom_refresh_indicator syncfusion_flutter_datepicker bb_core]
- calendar_timeline 3.0.4 [flutter flutter_localizations intl scrollable_positioned_list]
- collection 1.18.0
- confetti 0.8.0 [flutter vector_math]
- connectivity_plus 6.1.0 [flutter flutter_web_plugins connectivity_plus_platform_interface web meta nm collection]
- contacts_service 0.6.3 [flutter collection quiver]
- control 0.2.0 [flutter meta]
- convert 3.1.1 [typed_data]
- crypto 3.0.6 [typed_data]
- cupertino_icons 1.0.8
- device_info_plus 11.1.1 [device_info_plus_platform_interface ffi file flutter flutter_web_plugins meta web win32 win32_registry]
- dio 5.7.0 [async http_parser meta path dio_web_adapter]
- dio_smart_retry 6.0.0 [dio http_parser path]
- drift 2.21.0 [async convert collection meta stream_channel sqlite3 path stack_trace web]
- drift_db_viewer 2.1.0 [db_viewer drift flutter provider]
- equatable 2.0.5 [collection meta]
- firebase_analytics 11.3.4 [firebase_analytics_platform_interface firebase_analytics_web firebase_core firebase_core_platform_interface flutter]
- firebase_core 3.7.0 [firebase_core_platform_interface firebase_core_web flutter meta]
- firebase_messaging 15.1.4 [firebase_core firebase_core_platform_interface firebase_messaging_platform_interface firebase_messaging_web flutter meta]
- firebase_remote_config 5.1.4 [firebase_core firebase_core_platform_interface firebase_remote_config_platform_interface firebase_remote_config_web flutter]
- flutter 0.0.0 [characters collection material_color_utilities meta vector_math sky_engine]
- flutter_app_badger 1.5.0 [flutter]
- flutter_easyloading 3.0.5 [flutter flutter_spinkit]
- flutter_in_store_app_version_checker 1.9.0 [flutter meta http package_info_plus]
- flutter_local_notifications 18.0.0 [clock flutter flutter_local_notifications_linux flutter_local_notifications_platform_interface timezone]
- flutter_localizations 0.0.0 [flutter intl characters clock collection material_color_utilities meta path vector_math]
- flutter_markdown 0.7.4 [flutter markdown meta path]
- flutter_resource_calendar 2.2.0 [flutter flutter_localizations intl collection meta linked_scroll_controller bb_core bb_ui_kit]
- flutter_secure_storage 9.2.2 [flutter flutter_secure_storage_linux flutter_secure_storage_macos flutter_secure_storage_platform_interface flutter_secure_storage_web flutter_secure_storage_windows meta]
- flutter_speed_dial 7.0.0 [flutter]
- flutter_widgetkit 1.0.3 [flutter]
- get_it 7.6.7 [async collection]
- grouped_list 6.0.0 [flutter]
- in_app_review 2.0.9 [flutter in_app_review_platform_interface]
- intl 0.19.0 [clock meta path]
- l 5.0.0-pre.2 [meta]
- meta 1.15.0
- package_info_plus 8.1.1 [ffi flutter flutter_web_plugins http meta path package_info_plus_platform_interface web win32 clock]
- path 1.9.0
- path_provider 2.1.5 [flutter path_provider_android path_provider_foundation path_provider_linux path_provider_platform_interface path_provider_windows]
- permission_handler 11.3.1 [flutter meta permission_handler_android permission_handler_apple permission_handler_html permission_handler_windows permission_handler_platform_interface]
- platform_info 5.0.0 [meta web]
- quick_actions 1.0.8 [flutter quick_actions_android quick_actions_ios quick_actions_platform_interface]
- rive 0.13.15 [collection flutter flutter_web_plugins http meta plugin_platform_interface rive_common]
- rxdart 0.28.0
- sensors_plus 6.1.0 [flutter flutter_web_plugins sensors_plus_platform_interface]
- sentry_flutter 8.10.1 [flutter flutter_web_plugins sentry package_info_plus meta ffi file]
- share_plus 10.1.2 [cross_file meta mime flutter flutter_web_plugins share_plus_platform_interface file url_launcher_web url_launcher_windows url_launcher_linux url_launcher_platform_interface ffi web win32]
- shared_preferences 2.3.3 [flutter shared_preferences_android shared_preferences_foundation shared_preferences_linux shared_preferences_platform_interface shared_preferences_web shared_preferences_windows]
- showcaseview 3.0.0 [flutter]
- sqlite3_flutter_libs 0.5.26 [flutter]
- stack_trace 1.11.1 [path]
- stacked 3.4.3 [flutter get_it meta provider collection stacked_shared universal_io path]
- table_sticky_headers 2.0.5 [flutter]
- web 1.1.0

dev dependencies:
- auto_route_generator 9.0.0 [build source_gen analyzer path build_runner code_builder dart_style xml args glob package_config auto_route]
- build_runner 2.4.13 [analyzer args async build build_config build_daemon build_resolvers build_runner_core code_builder collection crypto dart_style frontend_server_client glob graphs http_multi_server io js logging meta mime package_config path pool pub_semver pubspec_parse shelf shelf_web_socket stack_trace stream_transform timing watcher web_socket_channel yaml]
- drift_dev 2.21.2 [charcode collection recase meta path json_annotation stream_transform args logging cli_util yaml io drift sqlite3 sqlparser analyzer analyzer_plugin source_span package_config pub_semver build build_resolvers build_config dart_style source_gen string_scanner]
- faker 2.2.0 [crypto]
- flutter_gen_runner 5.8.0 [flutter_gen_core build collection crypto glob path yaml]
- flutter_launcher_icons 0.14.1 [args checked_yaml cli_util image json_annotation path yaml]
- flutter_lints 4.0.0 [lints]
- flutter_native_splash 2.4.2 [args flutter flutter_web_plugins html image meta path universal_io xml yaml ansicolor]
- flutter_test 0.0.0 [flutter test_api matcher path fake_async clock stack_trace vector_math leak_tracker_flutter_testing async boolean_selector characters collection leak_tracker leak_tracker_testing material_color_utilities meta source_span stream_channel string_scanner term_glyph vm_service]
- integration_test 0.0.0 [flutter flutter_driver flutter_test path vm_service async boolean_selector characters clock collection fake_async file leak_tracker leak_tracker_flutter_testing leak_tracker_testing matcher material_color_utilities meta source_span stack_trace stream_channel string_scanner sync_http term_glyph test_api vector_math webdriver]
- mockito 5.4.4 [analyzer build code_builder collection dart_style matcher meta path source_gen test_api]
- plugin_platform_interface 2.1.8 [meta]
- pubspec_generator 4.1.0-pre.1 [analyzer build pub_semver yaml]
- shared_preferences_platform_interface 2.4.1 [flutter plugin_platform_interface]
- url_launcher_platform_interface 2.3.2 [flutter plugin_platform_interface]

dependency overrides:
- bb_core 2.4.0 [flutter flutter_localizations intl http flutter_secure_storage shared_preferences async path meta collection path_provider url_launcher]
- bb_http_client 3.9.0 [flutter flutter_localizations intl collection convert async meta path dio dio_smart_retry retry sentry_flutter l stack_trace bb_core]
- bb_ui_kit 3.1.0 [flutter flutter_localizations intl path meta async convert equatable collection platform_info url_launcher permission_handler mask_text_input_formatter flutter_svg flutter_slidable flutter_rating_bar flutter_inset_box_shadow cupertino_icons another_flushbar image_picker extended_image carousel_slider sticky_headers pull_down_button cached_network_image custom_refresh_indicator syncfusion_flutter_datepicker bb_core]
- calendar_timeline 3.0.4 [flutter flutter_localizations intl scrollable_positioned_list]
- flutter_resource_calendar 2.2.0 [flutter flutter_localizations intl collection meta linked_scroll_controller bb_core bb_ui_kit]

transitive dependencies:
- _fe_analyzer_shared 72.0.0 [meta]
- _flutterfire_internals 1.3.45 [collection firebase_core firebase_core_platform_interface flutter meta]
- _macros 0.3.2
- analyzer 6.7.0 [_fe_analyzer_shared collection convert crypto glob macros meta package_config path pub_semver source_span watcher yaml]
- analyzer_plugin 0.11.3 [analyzer collection dart_style pub_semver yaml]
- another_flushbar 1.12.30 [flutter]
- ansicolor 2.0.3
- archive 3.6.1 [crypto path]
- args 2.6.0
- boolean_selector 2.1.1 [source_span string_scanner]
- build 2.4.1 [analyzer async convert crypto glob logging meta package_config path]
- build_config 1.1.1 [checked_yaml json_annotation path pubspec_parse yaml]
- build_daemon 4.0.2 [built_collection built_value crypto http_multi_server logging path pool shelf shelf_web_socket stream_transform watcher web_socket_channel]
- build_resolvers 2.4.2 [analyzer async build collection convert crypto graphs logging package_config path pool pub_semver stream_transform yaml]
- build_runner_core 7.3.2 [async build build_config build_resolvers collection convert crypto glob graphs json_annotation logging meta package_config path pool timing watcher yaml]
- built_collection 5.1.1
- built_value 8.9.2 [built_collection collection fixnum meta]
- cached_network_image 3.4.1 [cached_network_image_platform_interface cached_network_image_web flutter flutter_cache_manager octo_image]
- cached_network_image_platform_interface 4.1.1 [flutter flutter_cache_manager]
- cached_network_image_web 1.3.1 [cached_network_image_platform_interface flutter flutter_cache_manager web]
- carousel_slider 5.0.0 [flutter]
- characters 1.3.0
- charcode 1.3.1
- checked_yaml 2.0.3 [json_annotation source_span yaml]
- cli_util 0.4.1 [meta path]
- clock 1.1.1
- code_builder 4.10.0 [built_collection built_value collection matcher meta]
- color 3.0.0
- connectivity_plus_platform_interface 2.0.1 [flutter meta plugin_platform_interface]
- cross_file 0.3.4+2 [meta web]
- csslib 1.0.0 [source_span]
- custom_refresh_indicator 4.0.1 [flutter meta]
- dart_style 2.3.6 [analyzer args collection path pub_semver source_span]
- dartx 1.2.0 [characters collection crypto meta path time]
- db_viewer 1.1.0 [flutter provider]
- dbus 0.7.10 [args ffi meta xml]
- decimal 3.0.2 [intl rational]
- device_info_plus_platform_interface 7.0.1 [flutter meta plugin_platform_interface]
- dio_web_adapter 1.0.1 [dio http_parser meta]
- extended_image 9.0.3 [extended_image_library flutter meta vector_math]
- extended_image_library 4.0.5 [crypto flutter http_client_helper js path path_provider web]
- fake_async 1.3.1 [clock collection]
- ffi 2.1.2
- file 7.0.0 [meta path]
- file_selector_linux 0.9.2+1 [cross_file file_selector_platform_interface flutter]
- file_selector_macos 0.9.4 [cross_file file_selector_platform_interface flutter]
- file_selector_platform_interface 2.6.2 [cross_file flutter http plugin_platform_interface]
- file_selector_windows 0.9.3+2 [cross_file file_selector_platform_interface flutter]
- firebase_analytics_platform_interface 4.2.6 [_flutterfire_internals firebase_core flutter meta plugin_platform_interface]
- firebase_analytics_web 0.5.10+3 [_flutterfire_internals firebase_analytics_platform_interface firebase_core firebase_core_web flutter flutter_web_plugins]
- firebase_core_platform_interface 5.3.0 [collection flutter flutter_test meta plugin_platform_interface]
- firebase_core_web 2.18.1 [firebase_core_platform_interface flutter flutter_web_plugins meta web]
- firebase_messaging_platform_interface 4.5.47 [_flutterfire_internals firebase_core flutter meta plugin_platform_interface]
- firebase_messaging_web 3.9.3 [_flutterfire_internals firebase_core firebase_core_web firebase_messaging_platform_interface flutter flutter_web_plugins meta web]
- firebase_remote_config_platform_interface 1.4.45 [_flutterfire_internals firebase_core flutter meta plugin_platform_interface]
- firebase_remote_config_web 1.7.3 [firebase_core firebase_core_web firebase_remote_config_platform_interface flutter flutter_web_plugins]
- fixnum 1.1.0
- flutter_cache_manager 3.4.1 [clock collection file flutter http path path_provider rxdart sqflite uuid]
- flutter_driver 0.0.0 [file flutter flutter_test fuchsia_remote_debug_protocol path meta vm_service webdriver async boolean_selector characters clock collection leak_tracker leak_tracker_flutter_testing leak_tracker_testing matcher material_color_utilities platform process source_span stack_trace stream_channel string_scanner sync_http term_glyph test_api vector_math]
- flutter_gen_core 5.8.0 [meta path yaml mime xml dartx color collection json_annotation glob dart_style args pub_semver vector_graphics_compiler image_size_getter]
- flutter_inset_box_shadow 1.0.8 [flutter]
- flutter_local_notifications_linux 5.0.0 [dbus ffi flutter flutter_local_notifications_platform_interface path xdg_directories]
- flutter_local_notifications_platform_interface 8.0.0 [flutter plugin_platform_interface]
- flutter_plugin_android_lifecycle 2.0.21 [flutter]
- flutter_rating_bar 4.0.1 [flutter]
- flutter_secure_storage_linux 1.2.1 [flutter flutter_secure_storage_platform_interface]
- flutter_secure_storage_macos 3.1.2 [flutter flutter_secure_storage_platform_interface]
- flutter_secure_storage_platform_interface 1.1.2 [flutter plugin_platform_interface]
- flutter_secure_storage_web 1.2.1 [flutter flutter_secure_storage_platform_interface flutter_web_plugins js]
- flutter_secure_storage_windows 3.1.2 [ffi flutter flutter_secure_storage_platform_interface path path_provider win32]
- flutter_slidable 3.1.1 [flutter]
- flutter_spinkit 5.2.1 [flutter]
- flutter_svg 2.0.10+1 [flutter http vector_graphics vector_graphics_codec vector_graphics_compiler]
- flutter_web_plugins 0.0.0 [flutter characters collection material_color_utilities meta vector_math]
- frontend_server_client 4.0.0 [async path]
- fuchsia_remote_debug_protocol 0.0.0 [process vm_service file meta path platform]
- glob 2.1.2 [async collection file path string_scanner]
- graphs 2.3.2 [collection]
- hashcodes 2.0.0
- html 0.15.4 [csslib source_span]
- http 1.2.2 [async http_parser meta web]
- http_client_helper 3.0.0 [http]
- http_multi_server 3.2.1 [async]
- http_parser 4.0.2 [collection source_span string_scanner typed_data]
- image 4.3.0 [archive meta xml]
- image_picker 1.1.2 [flutter image_picker_android image_picker_for_web image_picker_ios image_picker_linux image_picker_macos image_picker_platform_interface image_picker_windows]
- image_picker_android 0.8.12+11 [flutter flutter_plugin_android_lifecycle image_picker_platform_interface]
- image_picker_for_web 3.0.5 [flutter flutter_web_plugins image_picker_platform_interface mime web]
- image_picker_ios 0.8.12 [flutter image_picker_platform_interface]
- image_picker_linux 0.2.1+1 [file_selector_linux file_selector_platform_interface flutter image_picker_platform_interface]
- image_picker_macos 0.2.1+1 [file_selector_macos file_selector_platform_interface flutter image_picker_platform_interface]
- image_picker_platform_interface 2.10.0 [cross_file flutter http plugin_platform_interface]
- image_picker_windows 0.2.1+1 [file_selector_platform_interface file_selector_windows flutter image_picker_platform_interface]
- image_size_getter 2.1.3 [collection hashcodes]
- in_app_review_platform_interface 2.0.5 [flutter url_launcher plugin_platform_interface platform]
- io 1.0.4 [meta path string_scanner]
- js 0.6.7 [meta]
- json_annotation 4.9.0 [meta]
- leak_tracker 10.0.5 [clock collection meta path vm_service]
- leak_tracker_flutter_testing 3.0.5 [flutter leak_tracker leak_tracker_testing matcher meta]
- leak_tracker_testing 3.0.1 [leak_tracker matcher meta]
- linked_scroll_controller 0.2.0 [flutter]
- lints 4.0.0
- logging 1.2.0
- macros 0.1.2-main.4 [_macros]
- markdown 7.2.2 [args meta]
- mask_text_input_formatter 2.9.0 [flutter]
- matcher 0.12.16+1 [async meta stack_trace term_glyph test_api]
- material_color_utilities 0.11.1 [collection]
- mime 1.0.5
- nested 1.0.0 [flutter]
- nm 0.5.0 [dbus]
- octo_image 2.1.0 [flutter]
- package_config 2.1.0 [path]
- package_info_plus_platform_interface 3.0.1 [flutter meta plugin_platform_interface]
- path_parsing 1.0.1 [vector_math meta]
- path_provider_android 2.2.9 [flutter path_provider_platform_interface]
- path_provider_foundation 2.4.0 [flutter path_provider_platform_interface]
- path_provider_linux 2.2.1 [ffi flutter path path_provider_platform_interface xdg_directories]
- path_provider_platform_interface 2.1.2 [flutter platform plugin_platform_interface]
- path_provider_windows 2.3.0 [ffi flutter path path_provider_platform_interface]
- permission_handler_android 12.0.8 [flutter permission_handler_platform_interface]
- permission_handler_apple 9.4.5 [flutter permission_handler_platform_interface]
- permission_handler_html 0.1.3+2 [flutter flutter_web_plugins permission_handler_platform_interface web]
- permission_handler_platform_interface 4.2.2 [flutter meta plugin_platform_interface]
- permission_handler_windows 0.2.1 [flutter permission_handler_platform_interface]
- petitparser 6.0.2 [meta]
- platform 3.1.5
- pool 1.5.1 [async stack_trace]
- process 5.0.2 [file path platform]
- provider 6.1.2 [collection flutter nested]
- pub_semver 2.1.4 [collection meta]
- pubspec_parse 1.3.0 [checked_yaml collection json_annotation pub_semver yaml]
- pull_down_button 0.10.2 [flutter meta]
- quick_actions_android 1.0.15 [flutter quick_actions_platform_interface]
- quick_actions_ios 1.1.1 [flutter quick_actions_platform_interface]
- quick_actions_platform_interface 1.0.6 [flutter plugin_platform_interface]
- quiver 3.2.1 [matcher]
- rational 2.2.3
- recase 4.1.0
- retry 3.1.2
- rive_common 0.4.12 [collection ffi flutter flutter_web_plugins graphs http meta plugin_platform_interface web]
- scrollable_positioned_list 0.3.8 [flutter collection]
- sensors_plus_platform_interface 2.0.0 [flutter logging meta plugin_platform_interface]
- sentry 8.10.1 [http meta stack_trace uuid collection]
- share_plus_platform_interface 5.0.1 [cross_file flutter meta mime plugin_platform_interface path_provider uuid]
- shared_preferences_android 2.3.0 [flutter shared_preferences_platform_interface]
- shared_preferences_foundation 2.5.0 [flutter shared_preferences_platform_interface]
- shared_preferences_linux 2.4.0 [file flutter path path_provider_linux path_provider_platform_interface shared_preferences_platform_interface]
- shared_preferences_web 2.4.1 [flutter flutter_web_plugins shared_preferences_platform_interface web]
- shared_preferences_windows 2.4.0 [file flutter path path_provider_platform_interface path_provider_windows shared_preferences_platform_interface]
- shelf 1.4.1 [async collection http_parser path stack_trace stream_channel]
- shelf_web_socket 2.0.0 [shelf stream_channel web_socket_channel]
- sky_engine 0.0.99
- source_gen 1.5.0 [analyzer async build dart_style glob path source_span yaml]
- source_span 1.10.0 [collection path term_glyph]
- sprintf 7.0.0
- sqflite 2.3.3+1 [flutter sqflite_common path]
- sqflite_common 2.5.4 [synchronized path meta]
- sqlite3 2.4.6 [collection ffi meta path web]
- sqlparser 0.39.1 [meta collection source_span charcode]
- stacked_shared 1.4.1 [get_it]
- sticky_headers 0.3.0+2 [flutter]
- stream_channel 2.1.2 [async]
- stream_transform 2.1.0
- string_scanner 1.2.0 [source_span]
- sync_http 0.3.1
- syncfusion_flutter_core 27.1.53 [vector_math flutter]
- syncfusion_flutter_datepicker 27.1.53 [flutter intl syncfusion_flutter_core]
- synchronized 3.1.0+1
- term_glyph 1.2.1
- test_api 0.7.2 [async boolean_selector collection meta source_span stack_trace stream_channel string_scanner term_glyph]
- time 2.1.4 [clock]
- timezone 0.9.4 [path]
- timing 1.0.1 [json_annotation]
- typed_data 1.3.2 [collection]
- universal_io 2.2.2 [collection meta typed_data]
- url_launcher 6.3.1 [flutter url_launcher_android url_launcher_ios url_launcher_linux url_launcher_macos url_launcher_platform_interface url_launcher_web url_launcher_windows]
- url_launcher_android 6.3.8 [flutter url_launcher_platform_interface]
- url_launcher_ios 6.3.1 [flutter url_launcher_platform_interface]
- url_launcher_linux 3.2.0 [flutter url_launcher_platform_interface]
- url_launcher_macos 3.2.0 [flutter url_launcher_platform_interface]
- url_launcher_web 2.3.3 [flutter flutter_web_plugins url_launcher_platform_interface web]
- url_launcher_windows 3.1.2 [flutter url_launcher_platform_interface]
- uuid 4.4.2 [crypto sprintf meta fixnum]
- vector_graphics 1.1.11+1 [flutter http vector_graphics_codec]
- vector_graphics_codec 1.1.11+1
- vector_graphics_compiler 1.1.11+1 [args meta path_parsing xml vector_graphics_codec path]
- vector_math 2.1.4
- vm_service 14.2.5
- watcher 1.1.0 [async path]
- web_socket 0.1.6 [web]
- web_socket_channel 3.0.1 [async crypto stream_channel web web_socket]
- webdriver 3.0.3 [matcher path stack_trace sync_http]
- win32 5.5.3 [ffi]
- win32_registry 1.1.4 [ffi win32]
- xdg_directories 1.0.4 [meta path]
- xml 6.5.0 [collection meta petitparser]
- yaml 3.1.2 [collection source_span string_scanner]

```

</details>

**Comment by aabughazaleh** [comment]
In AppDelegete.swift try to put these lines in this order:
1) FirebaseApp.configure()
2) if #available(iOS 10.0, *) {
      UNUserNotificationCenter.current().delegate = self as? UNUserNotificationCenterDelegate
    }

**Comment by Lzyct** [comment]
> Solved.

How do you fix it?
